File: MyDailyBeat/API/GroupCreationForm.m
//
//  GroupCreationForm.m
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 10/31/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//

#import "GroupCreationForm.h"

@implementation GroupCreationForm

- (NSDictionary *) hobbiesField {
    return @{FXFormFieldTitle: @"Select Hobbies", FXFormFieldOptions: @[@"Books", @"Golf", @"Cars", @"Walking", @"Hiking", @"Wine", @"Woodworking", @"Online Card Games", @"Card Games", @"Online Games", @"Arts & Crafts", @"Prayer", @"Support Groups", @"Shopping", @"Travel", @"Local Field Trips", @"History", @"Sports"], FXFormFieldViewController: @"EVCGroupSettingsHobbiesSelectionTableViewController"};
}

- (NSArray *)extraFields
{
    return @[
             @{FXFormFieldTitle: @"Create", FXFormFieldHeader: @"", FXFormFieldAction: @"createGroup:"},
             ];
}



@end

File: MDB_Swift/MyDailyBeat/API/src/GroupCreationForm.swift
//  Converted with Swiftify v1.0.6242 - https://objectivec2swift.com/
//
//  GroupCreationForm.swift
//  MyDailyBeat
//
//  Created by Virinchi Balabhadrapatruni on 10/31/16.
//  Copyright © 2016 eVerveCorp. All rights reserved.
//
import Foundation
import FXForms
public class GroupCreationForm: NSObject, FXForm {
    public var groupName: String = ""
    public var hobbies: [Int: Bool] = [Int: Bool]()
    
    public func toggle(key: Int, value: Bool) {
        let values: [Bool] = hobbies.values.filter({ (val) -> Bool in
            return val
        })
        if value && values.count < 3 {
            self.hobbies[key] = value
        } else if !value {
            self.hobbies[key] = value
        }
    }
}
